tests/cpp/cell/test_tile_iterator.cu

```diff
@@ -0,0 +1,116 @@
+#include "common/test_utils.hpp"
+#include "types/mod.hpp"
+
+namespace tiledcuda {
+namespace testing {
+
+using namespace cell;
+using namespace cute;
+namespace tl = tile_layout;
+
+namespace {
+
+template <typename Element>
+__device__ void init_value(Element* data, int numel) {
+    for (int i = 0; i < numel; ++i) data[i] = static_cast<Element>(i);
+}
+
+template <typename Iterator>
+__global__ void test_tile_iterator() {
+    using DType = typename Iterator::Tile::DType;
+
+    extern __shared__ __align__(sizeof(double)) unsigned char buf_[];
+    auto* buf = reinterpret_cast<DType*>(buf_);
+    init_value(buf, Iterator::Tile::kNumel);
+
+    typename Iterator::Tile s_tile(buf);
+    printf("Shared tile:\n");
+    s_tile.dump_value();
+
+    Iterator tiles(buf);
+
+    printf("Iterate over rows.\n\n");
+    for (int i = 0; i < Iterator::sc0; ++i) {
+        printf("Iteration-[%d, _]:\n", i);
+        tiles(i, _).to_tile().dump_value();
+        printf("\n");
+    }
+
+    printf("Iterate over columns.\n\n");
+    for (int j = 0; j < Iterator::sc1; ++j) {
+        printf("Iteration-[_, %d]:\n", j);
+        tiles(_, j).to_tile().dump_value();
+        printf("\n");
+    }
+
+    printf("Iterate over rows and columns.\n\n");
+    for (int i = 0; i < Iterator::sc0; ++i) {
+        for (int j = 0; j < Iterator::sc1; ++j) {
+            printf("Iteration-[%d, %d]:\n", i, j);
+            tiles(i, j).dump_value();
+            printf("\n");
+        }
+    }
+
+    printf("Another way to iterate over rows and columns.\n\n");
+    for (int i = 0; i < Iterator::sc0; ++i) {
+        auto cols = tiles(i, _);
+
+        printf("\n");
+        for (int j = 0; j < decltype(cols)::sc1; ++j) {
+            printf("Iteration-[%d, %d]:\n", i, j);
+            cols(j).dump_value();
```

Comment:
Slicing a 2D iterator will return a 2D iterator with one dimension reduced to 1, resulting in a 1D iterator. This 1D iterator can be indexed using a 1D array index.

In the current implementation, a 2D iterator with any of its dimensions being 1 can be indexed using a 1D index.